lisa/tools/mdadm.py

```diff
@@ -47,3 +44,86 @@ def stop_raid(
         volume_name: str = "/dev/md0",
     ) -> None:
         self.run(f"--stop {volume_name}", force_run=True, sudo=True)
+
+    @classmethod
+    def _windows_tool(cls) -> Optional[Type[Tool]]:
+        return WindowsMdadm
+
+    def _install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages("mdadm")
+        return self._check_exists()
+
+
+class WindowsMdadm(Mdadm):
+    @property
+    def command(self) -> str:
+        return "powershell"
+
+    def _check_exists(self) -> bool:
+        return True
+
+    def create_raid(
+        self,
+        disk_list: List[str],
+        level: int = 0,
+        volume_name: str = "Raid0-Disk",
+        chunk_size: int = 0,
+        pool_name: str = "Raid0-Pool",
+    ) -> None:
+        powershell = self.node.tools[PowerShell]
+
+        # create pool
+        # TODO: add support for higher raid types and chunk sizes
+        self._create_pool(pool_name)
+
+        # create new virtual disk
+        self.node.tools[HyperV].create_virtual_disk(volume_name, pool_name)
+
+        # set raid disk offline
+        raid_disk_id = int(
+            powershell.run_cmdlet(
+                "(Get-Disk "
+                f"| Where-Object {{$_.FriendlyName -eq '{volume_name}'}}).Number",
+                force_run=True,
+            ).strip()
+        )
+        powershell.run_cmdlet(
+            f"Set-Disk {raid_disk_id} -IsOffline $true", force_run=True
+        )
+
+    def stop_raid(
+        self, volume_name: str = "Raid0-Disk", pool_name: str = "Raid0-Pool"
+    ) -> None:
+        # delete virtual disk if it exists
+        self.node.tools[HyperV].delete_virtual_disk(volume_name)
+
+        # delete storage pool
+        self._delete_pool(pool_name)
+
+    def _exists_pool(self, pool_name: str) -> bool:
+        output = self.node.tools[PowerShell].run_cmdlet(
+            f"Get-StoragePool -FriendlyName {pool_name}",
+            fail_on_error=False,
+            force_run=True,
+        )
+        return output.strip() != ""
+
+    def _delete_pool(self, pool_name: str) -> None:
+        if self._exists_pool(pool_name):
+            self.node.tools[PowerShell].run_cmdlet(
+                f"Remove-StoragePool -FriendlyName {pool_name} -confirm:$false",
+                force_run=True,
+            )
+
+    def _create_pool(self, pool_name: str) -> None:
+        # delete pool if exists
+        self._delete_pool(pool_name)
+
+        # create pool
+        self.node.tools[PowerShell].run_cmdlet(
+            "$disks = Get-PhysicalDisk -CanPool  $true; New-StoragePool "
```

Comment:
Is the pool the same as raid?

Reply:
Pool is basically a set of disks.  We first create pool and then setup raid0 enabled disk from it

Reply:
Which command creates the raid disk? I cannot find command specified raid.

Reply:
`New-VirtualDisk` is the function name. The entire logic as defined in v2 is here: https://github.com/microsoft/lisa/blob/24061a4288a02bd223beb878344d481bc0d77e3e/Testscripts/Windows/NESTED-HYPERV-STORAGE-PERF.ps1#L47-L66

Reply:
The Windows Storage pool is not the RAID technology, but they solve similar problems. Please check the test logic, if the storage pool is ok for the test logic.

Reply:
We need to add parameter `-NumberOfColumns 2` to strip it across two disks. I will add it in the next commit